# Laboratorio n° 4 - Circuito 1

Se iguala la expresión de la realimentación $T(s)$ con FT de segundo orden.

$$ T(s)= \frac{k N_{FF}}{D-kN_{FF}}= \frac{ \frac{k}{(RC)^{2}} }{ s^{2}+ \frac{(3-k)}{RC}s + (\frac{1}{RC})^{2} } $$

Para el filtro pasabajo se tiene lo siguiente:

$$ T(s)= \frac{3.284.10^7}{s^{2}+ 3184 \, s+ 6.632.10^{7}}= \frac{k\omega^{2}_{p}}{s^{2}+ \frac{\omega_{p}}{Q_{p}}s+\omega^{2}_{p}} $$

Se tiene:

$$ \frac{\omega_{p}}{Q_{p}}= \frac{(3-k)}{RC}= 3184 $$

$$ \omega_{p}= \frac{1}{RC}= \sqrt{66.32.10^{6}}= 8143 $$

$$ k\omega^{2}_p= \frac{k}{(RC)^{2}}= 32840000 $$

In [138]:
import sympy as sym

R, C, k, r1, r2= sym.symbols('R, C, k, r_{1}, r_{2}')

# Filtro pasabajo
equ1= sym.Eq((3-k)/(R*C), 3184)
equ2= sym.Eq(1/(R*C), 8143)
equ3= sym.Eq(k/(R*C)**2, 32840000)
equ_k= sym.Eq(k, 1+(r2/r1))

# Despejo equ2

# Establezco C= 100nF
valor_C= 100e-9

s_R= sym.solve(equ2, C)[0]
Rc= s_R.subs(R, valor_C)
print('C= ', valor_C)
print('R=', Rc)

s_k= sym.solve(equ1, k)
#print('k= ', s_k[0])
valor_k= s_k[0].subs({C: valor_C, R: Rc})
print('k= ', valor_k)

# Despejo r1 de equ_k
equ_r1= sym.solve(equ_k, r1)

# Establezco r2= 10Kohm
valor_r1= equ_r1[0].subs({r2: 10e3, k: valor_k})
print('r1= ', valor_r1)

C=  1e-07
R= 1228.04863072578
k=  2.60898931597691
r1=  6215.08166692108


Ganancia debe ser K=0.49 (ver LABN4_SRA), se debe compensar ya que obtuvimos K= 2.61 (obtenido) se debe agregar otra ganancia de 0.19 entonces K_teórico= 0.49= 2.61*0.19.

In [147]:
Ra, Rb= sym.symbols('R_{A}, R_{B}')

equ4= sym.Eq(Ra, R/0.19)

# Valor Ra
valor_Ra= equ4.subs(R, Rc)

# Valor Rb
equ5= sym.Eq(R, (Ra*Rb)/(Ra+Rb))
s_Rb= sym.solve(equ5, Rb)

valor_Rb= s_Rb[0].subs({Ra: valor_Ra.rhs, R: Rc})
print('Rb= ', valor_Rb)

Rb=  1516.10942064911
